### ISSUE TITLE
Getting SSM Parameter name instead of SSM Parameter value with aws-cdk and cdklocal

### ISSUE THREAD

**Issue by tinaboyce** [localstack/localstack#3738]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

Using AWS CDK, cdklocal and cdk context, trying to obtain an SSM Parameter value using SSM Parameter name from CDK Context. The issue is `ssm.StringParameter.valueForStringParameter` returns the SSM Parameter name instead of the SSM Parameter value.

cdk.json, in context section, add:
```
{
  ...
  "context": {
    ...
    "regionPath": "/region"
  }
}
```

Value from an AWS SSM get-parameter call
```
➜  cdk-testing-playground git:(master) ✗ aws --endpoint-url=http://localhost:4566 ssm get-parameter --name "/region"
{
    "Parameter": {
        "Name": "region",
        "Type": "String",
        "Value": "sydney",
        "Version": 1,
        "LastModifiedDate": "2021-03-17T16:21:56.208000+11:00",
        "ARN": "arn:aws:ssm:us-east-1:1234567890:parameter/region"
    }
}
➜  cdk-testing-playground git:(master) ✗
```

main stack code (snipped)
```
    const regionSSMPath = this.node.tryGetContext("regionPath");
    const region = ssm.StringParameter.valueForStringParameter(
      this,
      regionSSMPath
    );
    new cdk.CfnOutput(this, "ssmParameterValueRegion", { value: region });
```

Output from the code above (snipped) after running `cdklocal deploy`. I'm expecting 'sydney', not '/region'
```
Outputs:
CdkTestingPlaygroundStack.ssmValueRegion = /region
```

## Expected behavior

`ssm.StringParameter.valueForStringParameter` to return "sydney" (the SSM Parameter value)

## Actual behavior

`ssm.StringParameter.valueForStringParameter` returns "/region" (the SSM Parameter name)

# Steps to reproduce

## Command used to start LocalStack
`docker-compose up`
```
version: '2.1'

services:
  localstack:
    container_name: "${LOCALSTACK_DOCKER_NAME-localstack_main}"
    image: localstack/localstack:latest
    network_mode: bridge
    ports:
      - "4566:4566"
      - "4571:4571"
      - "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - SERVICES=${SERVICES- }
      - DEBUG=${DEBUG- }
      - DATA_DIR=${DATA_DIR- }
      - PORT_WEB_UI=${PORT_WEB_UI- }
      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
      - DOCKER_HOST=unix:///var/run/docker.sock
      - HOST_TMP_FOLDER=${TMPDIR}
    volumes:
      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

1. Seeding SSM after localstack is ready
  ```
  aws --endpoint-url=http://localhost:4566 \
      ssm put-parameter \
      --name "/region" \
      --type "String" \
      --value "sydney"
  ```
2. `npm run build`
3. `cdklocal bootstrap`
4. `cdklocal deploy`

Update: 
- `ssm.StringParameter.valueFromLookup(this,regionSSMPath)` works, just not `ssm.StringParameter.valueForStringParameter(this,regionSSMPath)`

**Comment by tinaboyce** [comment]
Are there any update on this?

**Comment by tinaboyce** [comment]
@usmangani1 are there any update on this?

**Comment by tinaboyce** [comment]
Updated original post to say `ssm.StringParameter.valueFromLookup(this, regionSSMPath);` does return the value but I need to use `ssm.StringParameter.valueForStringParameter(this, regionSSMPath);` instead.

**Comment by computerwizjared** [comment]
I am running into this as well. We use SSM parameters a lot in our CDK projects so it would be very useful to get this fixed.

**Comment by dominikschubert** [comment]
Fixed in #4387

SSM parameters should now be properly resolved when deploying. Let me know if you have any further related issues. :+1:
